File: gamerecs-back/src/main/java/com/gamerecs/back/dto/ProfileResponseDto.java
package com.gamerecs.back.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ProfileResponseDto {
    private String username;
    private String email;
    private String profilePictureUrl;
    private String bio;
    private boolean emailVerified;
} 

File: gamerecs-front/src/app/models/profile-response.dto.ts
export interface ProfileResponseDto {
  username: string;
  email: string;
  profilePictureUrl?: string;
  bio?: string;
  emailVerified: boolean;
  steamProfileId?: string;
  steamCredentialsSet?: boolean;
}

export interface UpdateProfileRequest {
  username: string;
  profilePictureUrl?: string;
  bio?: string;
  steamApiKey?: string;
  steamProfileId?: string;
} 
